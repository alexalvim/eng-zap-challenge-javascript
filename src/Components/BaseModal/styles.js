import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';

export const OverlayWrapper = styled.div`
  align-items: center;
  background-color: ${Colors.overlayColor}
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 5;
`;

export const ModalContent = styled.div`
  background-color: ${Colors.lighterColor};
  border-radius: 5px;
  height: 100%;
  position: relative;
  width: 100%;

  @media(min-width: ${Spaces.tabletBreakpoint}) {
    height: auto;
    max-height: 600px;
    max-width: 500px;
  }
`;

export const CloseButton = styled.button`
  background-color: ${Colors.lighterColor};
  border: none;
  border-radius: 50%;
  box-shadow: 1px 3px 3px rgba(0,0,0,0.5);
  cursor: pointer;
  height: 30px;
  padding: ${Spaces.half};
  position: absolute;
  right: ${Spaces.base};
  top: ${Spaces.base};
  width: 30px;
  z-index: 1;
`;

export const closeButtonIconStyles = {
  color: Colors.darkerColor
}