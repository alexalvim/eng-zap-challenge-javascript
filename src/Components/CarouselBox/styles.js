import styled from 'styled-components';

import Spaces from '../../spaces';
import Typo from '../../typo';

export const ContentWrapper = styled.div`
  border-radius: 5px;
  box-shadow: 0px 3px 5px rgba(0,0,0,0.2);
  display: flex;
  flex-direction: column;
  margin: ${Spaces.double} 0;
  overflow: hidden;

  @media(min-width: ${Spaces.tabletBreakpoint}){
    flex-direction: row;
  }
`;

export const CarouselContainer = styled.div`
  width: 100%;

  .slick-arrow:before {
    opacity: 1;
    text-shadow: 1px 3px 3px rgba(0,0,0,0.5);
  }

  .slick-arrow.slick-disabled:before {
    opacity: .5;
  }

  .slick-next {
    right: 5px;
    z-index: 1;
  }

  .slick-prev {
    left: 5px;
    z-index: 1;
  }

  @media(min-width: ${Spaces.tabletBreakpoint}){
    width: 350px;
  }
`;

export const ImageContainer = styled.div`
  align-items: center;
  display: flex !important;
  height: 300px;
  justify-content: center;
  overflow: hidden;
  position: relative;

  > img {
    width: auto;
    height: 300px;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
  }

  @media(min-width: ${Spaces.tabletBreakpoint}){
    max-width: 350px;
  }
`;

export const TextHolder = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${Spaces.base};
`;

export const titleWrapperStyles = `
  font-size: ${Typo.large};
  margin-bottom: ${Spaces.base};
`;