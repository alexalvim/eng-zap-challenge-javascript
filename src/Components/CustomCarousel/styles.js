import styled from 'styled-components';

export const CarouselContainer = styled.div`
  width: 100%;

  .slick-arrow:before {
    opacity: 1;
    text-shadow: 1px 3px 3px rgba(0,0,0,0.5);
  }

  .slick-arrow.slick-disabled:before {
    opacity: .5;
  }

  .slick-next {
    right: 5px;
    z-index: 1;
  }

  .slick-prev {
    left: 5px;
    z-index: 1;
  }
`;

export const ImageContainer = styled.div`
  align-items: center;
  display: flex !important;
  justify-content: center;
  overflow: hidden;
  position: relative;

  > img {
    width: auto;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
  }
  ${({height}) => height &&
  `
    height: ${height}px;

    > img {
      height: ${height}px;
    }
  `}
`;