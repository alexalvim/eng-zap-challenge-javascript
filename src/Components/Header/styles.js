import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';

export const ContentWrapper = styled.div`
  background-color: ${Colors.lighterColor};
  color: ${Colors.lighterColor};
  padding: ${Spaces.double};
  width: 100%;

  ${({activePortal}) => activePortal && `border-bottom: solid 2px ${Colors.portalMainColor[activePortal]};`}
`;

export const Container = styled.div`
  margin: 0 auto;
  max-width: ${Spaces.containerMaxWidth};
  width: 100%;
`;

export const Logo = styled.img`
  height: 40px;
`;