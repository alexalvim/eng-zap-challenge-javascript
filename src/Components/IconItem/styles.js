import styled from 'styled-components';

import Colors from '../../colors';
import Typo from '../../typo'
import Spaces from '../../spaces';

export const ContentWrapper = styled.div`
  align-items: center;
  display: flex;
`;

export const Label = styled.span`
  color: ${Colors.darkerColor};
  font-size: ${Typo.medium};
`;

export const iconStyles = {
  marginRight: Spaces.base,
}