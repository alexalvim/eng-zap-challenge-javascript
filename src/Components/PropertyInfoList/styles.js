import styled from 'styled-components';

import Spaces from '../../spaces';

export const List = styled.ul`
  > li {
    margin-bottom: ${Spaces.base};
  }
`;