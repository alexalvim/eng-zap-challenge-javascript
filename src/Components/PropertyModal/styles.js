import styled from 'styled-components';

import Spaces from '../../spaces';
import Typo from '../../typo';
import Colors from '../../colors';

export const TextWrapper = styled.div`
  padding: ${Spaces.double};
`;

export const Title = styled.h2`
  font-size: ${Typo.large};
  margin-bottom: ${Spaces.double};

  ${({activePortal}) => activePortal && `color: ${Colors.portalMainColor[activePortal]}`}
`;

export const Price = styled.span`
  color: ${Colors.darkerColor};
  display: inline-block;
  font-size: ${Typo.medium};
  margin-top: ${Spaces.double};
`;