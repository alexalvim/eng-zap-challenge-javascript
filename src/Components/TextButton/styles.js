import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';
import Typo from '../../typo';

export const Button = styled.button`
  background: none;
  border: none;
  cursor: pointer;
  font-size: ${Typo.medium}
  font-weight: 500;
  padding: 0;

  &:hover {
    text-decoration: underline;
  }

  ${({activePortal}) => activePortal && `color: ${Colors.portalMainColor[activePortal]};`}
`;