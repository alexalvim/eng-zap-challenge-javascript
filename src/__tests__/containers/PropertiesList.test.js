import React from 'react';
import { shallow } from 'enzyme';

import { PropertiesList } from '../../containers/PropertiesList';
import { SimpleText } from '../../containers/PropertiesList/styles';

const mockedProperties = {
  errorMessage: '',
  isLoading: false,
  activeProperties: [{
    id: 1,
    pricingInfos: {
      businessType: 'SALE'
    }
  }]
}

describe('Testing PropertiesList', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <PropertiesList
        properties={mockedProperties}/>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should show error message when errorMessage exists', () => {
    const wrapper = shallow(
      <PropertiesList
        properties={{...mockedProperties, errorMessage: 'error'}}/>
    );

    expect(wrapper.find(SimpleText).text()).toEqual('Erro: error');
  });

  it('should show loadingMessage when isLoading is true', () => {
    const wrapper = shallow(
      <PropertiesList
        properties={{...mockedProperties, isLoading: true}}/>
    );

    expect(wrapper.find(SimpleText).text()).toEqual('Carregando ...');
  });

  it('should call getPropertiesRequest when call componentDidMount', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(
      <PropertiesList
        getPropertiesRequest={mockFunction}
        properties={{...mockedProperties, activeProperties: [] }}/>
    );

    wrapper.instance().componentDidMount();
    expect(mockFunction).toHaveBeenCalled();
  });
});

describe('Testing PropertiesList handleFunctions', () => {
  it('should call changeActivePortal when call handleClickChangePortal', () => {
    const mockFunction = jest.fn()
    const mockedArgument = 'foo';
    const wrapper = shallow(
      <PropertiesList
        changeActivePortal={mockFunction}
        properties={mockedProperties}/>
    );

    wrapper.instance().handleClickChangePortal(mockedArgument);
    expect(mockFunction).toHaveBeenCalledWith(mockedArgument);
  });

  it('should call changeActivePortal when call handleClickSeeMore', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(
      <PropertiesList
        seeMoreProperties={mockFunction}
        properties={mockedProperties}/>
    );

    wrapper.instance().handleClickSeeMore();
    expect(mockFunction).toHaveBeenCalled();
  });

  it('should change state value and call selectActiveProperty when call handleClickProperty', () => {
    const mockFunction = jest.fn()
    const mockedProperty ='property';
    const wrapper = shallow(
      <PropertiesList
        selectActiveProperty={mockFunction}
        properties={mockedProperties}/>
    );
    wrapper.instance().state.openPropertyModal = false;
    wrapper.instance().handleClickProperty(mockedProperty);

    expect(wrapper.instance().state.openPropertyModal).toEqual(true);
    expect(mockFunction).toHaveBeenCalledWith(mockedProperty);
  });

  it('should change state value and call selectActiveProperty when call handleClosePropertyModal', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(
      <PropertiesList
        selectActiveProperty={mockFunction}
        properties={mockedProperties}/>
    );
    wrapper.instance().state.openPropertyModal = true;
    wrapper.instance().handleClosePropertyModal();

    expect(wrapper.instance().state.openPropertyModal).toEqual(false);
    expect(mockFunction).toHaveBeenCalled();
  });
});