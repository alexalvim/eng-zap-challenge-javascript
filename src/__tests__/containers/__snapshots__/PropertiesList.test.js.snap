// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PropertiesList should render correctly 1`] = `ShallowWrapper {}`;
